core/domain/collection_domain_test.py

```diff
@@ -318,30 +329,64 @@ def test_add_delete_node(self):
         collection.delete_node('test_exp')
         self.assertEqual(len(collection.nodes), 0)
 
-    def test_skills_property(self):
+    def test_add_skill(self):
+        """Test that add_skill fails in the correct situations."""
         collection = collection_domain.Collection.create_default_collection(
-            '0')
+            'exp_id')
+        self.assertEqual(collection.skills, {})
 
-        self.assertEqual(collection.skills, [])
+        # Add skills
+        collection.add_skill('skillname1')
+        self.assertEqual(collection.skills.keys(), ['skill0'])
+        self.assertEqual(collection.skills['skill0'].name, 'skillname1')
 
-        collection.add_node('exp_id_0')
-        collection.add_node('exp_id_1')
-        collection.get_node('exp_id_0').update_acquired_skills(
-            ['skill0a'])
-        collection.get_node('exp_id_1').update_prerequisite_skills(
-            ['skill0a'])
-        collection.get_node('exp_id_1').update_acquired_skills(
-            ['skill1b', 'skill1c'])
+        collection.add_skill('skillname2')
+        self.assertEqual(sorted(collection.skills.keys()), ['skill0', 'skill1'])
+
+        # Names should be unique
+        with self.assertRaisesRegexp(
+            ValueError, 'Skill with name "skillname1" already exists.'):
+            collection.add_skill('skillname1')
 
-        self.assertEqual(collection.skills, ['skill0a', 'skill1b', 'skill1c'])
+        # Delete a skill
+        collection.delete_skill('skill1')
+        self.assertEqual(collection.skills.keys(), ['skill0'])
 
-        # Skills should be unique, even if they are duplicated across multiple
-        # acquired and prerequisite skill lists.
-        collection.add_node('exp_id_2')
-        collection.get_node('exp_id_2').update_acquired_skills(
-            ['skill0a', 'skill1c'])
-        self.assertEqual(collection.skills, ['skill0a', 'skill1b', 'skill1c'])
+        # Should raise error if ID is not found
+        with self.assertRaisesRegexp(
+            ValueError, 'Skill with ID "skill1" does not exist.'):
+            collection.delete_skill('skill1')
 
+        # New IDs should skip deleted IDs
```

Comment:
Perhaps you need a test which tests both delete and adding at the same time, e.g.

``test_add_skill_after_deletion_succeeds``